ts/torch_handler/base_handler.py

```diff
@@ -25,46 +21,74 @@
 else:
     PROFILER_AVAILABLE = False
 
+
+logger = logging.getLogger(__name__)
+
+
 try:
     import torch_xla.core.xla_model as xm
 
-    TORCHXLA_AVAILABLE = True
+    XLA_AVAILABLE = True
 except ImportError as error:
-    TORCHXLA_AVAILABLE = False
+    XLA_AVAILABLE = False
 
 
-logger = logging.getLogger(__name__)
-
-
-# Possible values for backend in utils.py
-def check_pt2_enabled():
-    try:
-        import torch._dynamo
-
-        pt2_enabled = True
-        if torch.cuda.is_available():
-            # If Ampere enable tensor cores which will give better performance
-            # Ideally get yourself an A10G or A100 for optimal performance
-            if torch.cuda.get_device_capability() >= (8, 0):
-                torch.backends.cuda.matmul.allow_tf32 = True
-    except ImportError as error:
-        logger.warning(
-            "dynamo/inductor are not installed. \n For GPU please run pip3 install numpy --pre torch[dynamo] --force-reinstall --extra-index-url https://download.pytorch.org/whl/nightly/cu117 \n for CPU please run pip3 install --pre torch --extra-index-url https://download.pytorch.org/whl/nightly/cpu"
-        )
-        pt2_enabled = False
-    return pt2_enabled
+if packaging.version.parse(torch.__version__) >= packaging.version.parse("2.0.0"):
+    PT2_AVAILABLE = True
+    if torch.cuda.is_available():
+        # If Ampere enable tensor cores which will give better performance
+        # Ideally get yourself an A10G or A100 for optimal performance
+        if torch.cuda.get_device_capability() >= (8, 0):
+            torch.backends.cuda.matmul.allow_tf32 = True
+            logger.info("Enabled tensor cores")
+else:
+    logger.warning(
+        f"Your torch version is {torch.__version__} which does not support torch.compile"
+    )
+    PT2_AVAILABLE = False
 
 
-ipex_enabled = False
 if os.environ.get("TS_IPEX_ENABLE", "false") == "true":
     try:
         import intel_extension_for_pytorch as ipex
 
-        ipex_enabled = True
+        IPEX_AVAILABLE = True
     except ImportError as error:
         logger.warning(
             "IPEX is enabled but intel-extension-for-pytorch is not installed. Proceeding without IPEX."
         )
+        IPEX_AVAILABLE = False
+else:
+    IPEX_AVAILABLE = False
+
+
+try:
+    import onnxruntime as ort
+    import psutil
+
+    logger.info("ONNX enabled")
+    ONNX_AVAILABLE = True
+except ImportError as error:
+    logger.warning("proceeding without onnxruntime")
+    ONNX_AVAILABLE = False
+
+
+def setup_ort_session(model_pt_path):
+    providers = (
+        ["CUDAExecutionProvider", "CPUExecutionProvider"]
```

Comment:
wonder if it should be  "CUDAExecutionProvider" if `self.map_location == "cuda"`?

Reply:
Oh yeah good point, I'm not too happy with the state of our ONNX tests at this point so maybe this is a good opportunity to revisit

Reply:
ACtually just checked this is not needed, ORT will read in the list of providers from left to right and pick the first available one, so if cuda is not available cpu will be picked